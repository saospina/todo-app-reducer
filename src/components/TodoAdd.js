import React from 'react'
import { useForm } from '../hooks/useForm';
import { TextField, Button } from '@material-ui/core';


export const TodoAdd = ({ handleAddTodo }) => {

    const [{ description }, handleInputChange, reset] = useForm({
        description: ''
    });

    const handleSubmit = (e) => {
        e.preventDefault();

        if (description.trim().length <= 1) {
            return;
        }

        const newTodo = {
            id: new Date().getTime(),
            desc: description,
            done: false
        };

        handleAddTodo(newTodo);
        reset();

    }


    return (
        <>
            <h4>Adding TODO's</h4>
            <hr />

            <form onSubmit={handleSubmit}>
                <TextField
                    name="description"
                    label="Add a new task"
                    type="search"
                    autoComplete="off"
                    variant="outlined"
                    value={description}
                    onChange={handleInputChange}

                />

                <Button
                    style={{marginLeft: '20px'}}
                    variant="contained"
                    size="large"
                    type="submit"
                    color="primary"
                >
                    Add

                </Button>

            </form>

        </>
    )
}