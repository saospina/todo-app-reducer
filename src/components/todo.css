.complete {
    text-decoration: line-through;
}

.add-task {
    margin-left: 20px;
}