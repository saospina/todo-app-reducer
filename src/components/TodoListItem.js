import React from 'react';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
import ListItemText from '@material-ui/core/ListItemText';
import Checkbox from '@material-ui/core/Checkbox';
import HighlightOffIcon from '@material-ui/icons/HighlightOff';

import { makeStyles } from '@material-ui/core/styles';
import './todo.css';



const useStyles = makeStyles((theme) => ({
    root: {
        width: '100%',
        maxWidth: 360,
        backgroundColor: theme.palette.background.paper,
    },
}));



export const TodoListItem = ({ todo, index, handleDelete, handleToggle }) => {

    const complete = {
        textDecoration: todo.done ? 'line-through' : 'none'
    }
    const classes = useStyles();

    const labelId = `checkbox-list-secondary-label-${todo.id}`;

    return (
        <List dense className={classes.root}>
            <ListItem
                key={todo.id} button
                style={complete}>
                <ListItemText id={labelId} primary={`${index + 1}. ${todo.desc}`} />
                <HighlightOffIcon
                    color="secondary"
                    onClick={() => handleDelete(todo.id)}
                />
                <ListItemSecondaryAction>
                    <Checkbox
                        color="primary"
                        edge="end"
                        onChange={() => handleToggle(todo.id)}
                        checked={todo.done}
                        inputProps={{ 'aria-labelledby': labelId }}
                    />
                </ListItemSecondaryAction>
            </ListItem>

        </List>
    )
}