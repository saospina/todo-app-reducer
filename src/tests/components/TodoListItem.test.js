import react from 'react';
import { shallow } from 'enzyme';
import { TodoListItem } from '../../components/TodoListItem';
import { demoTodos } from '../fixtures/demoTodos';

describe('Tests for TodoListItem componentw', () => {
    const handleDelete = jest.fn();
    const handleToggle = jest.fn();
    const wrapper = shallow(
        <TodoListItem
            todo={demoTodos[0]}
            index={0}
            handleDelete={handleDelete}
            handleToggle={handleToggle}
        />);

    test('should be shown correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should handleDelete function', () => {
        wrapper.find('HighlightOffIcon').simulate('click')
        expect(handleDelete).toHaveBeenCalledWith(demoTodos[0].id)
    })


})
