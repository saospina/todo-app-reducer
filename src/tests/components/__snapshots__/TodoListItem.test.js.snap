// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for TodoListItem componentw should be shown correctly 1`] = `
<WithStyles(ForwardRef(List))
  className="makeStyles-root-1"
  dense={true}
>
  <WithStyles(ForwardRef(ListItem))
    button={true}
    className="false"
    key="1"
  >
    <WithStyles(ForwardRef(ListItemText))
      id="checkbox-list-secondary-label-1"
      primary="Learn react js"
    />
    <Memo(ForwardRef(HighlightOffIcon))
      color="secondary"
      onClick={[Function]}
    />
    <WithStyles(ForwardRef(ListItemSecondaryAction))>
      <WithStyles(ForwardRef(Checkbox))
        checked={false}
        color="primary"
        edge="end"
        inputProps={
          Object {
            "aria-labelledby": "checkbox-list-secondary-label-1",
          }
        }
        onChange={[Function]}
      />
    </WithStyles(ForwardRef(ListItemSecondaryAction))>
  </WithStyles(ForwardRef(ListItem))>
</WithStyles(ForwardRef(List))>
`;
